week2/no1/src/pasta.js

```diff
@@ -0,0 +1,172 @@
+      modalHashtagItem.className = "modal_hashtag_item";
+
+      modalHashtagList.appendChild(modalHashtagItem);
+    });
+
+    const closeIconWrapper = document.createElement("span");
+    closeIconWrapper.className = "close_icon_button";
+
+    closeIconWrapper.addEventListener("click", () => {
+      modal.style.display = "none";
+    });
+
+    const closeIcon = document.createElement("i");
+    closeIcon.className = "fa-solid fa-xmark";
+
+    const image = document.createElement("img");
+    image.alt = menu.name;
+    image.src = menu.image_url;
+
+    const icon = document.createElement("i");
+    icon.className = "fa-solid fa-heart";
+
+    plusIconWrapper.appendChild(plusIcon);
+
+    closeIconWrapper.appendChild(closeIcon);
+
+    modal.appendChild(modalContent);
+    modalContent.appendChild(modalHashtagList);
+    modalContent.appendChild(closeIconWrapper);
+    card.appendChild(modal);
+
+    card.appendChild(title);
+    card.appendChild(hashtagContainer);
+    hashtagContainer.appendChild(hashtagList);
+    hashtagContainer.appendChild(plusIconWrapper);
+    card.appendChild(image);
+    card.appendChild(icon);
+
+    menuArea.appendChild(card);
+  });
+};
+
+//카테고리 선택 파트
+
+const categoryButton = document.querySelectorAll("#type_list > label input");
+const categoryList = ["all"];
+
+categoryButton.forEach((button) => {
+  button.addEventListener("click", (event) => {
+    const category = event.target.value;
+    // console.log("Clicked!!", category);
+    if (button.checked) {
+      if (!categoryList.includes(category)) {
+        categoryList.push(category);
+      }
+    }
+    displaySelectedCategory();
+    displayMenus();
+  });
+});
+
+const displaySelectedCategory = () => {
+  const categoryArea = document.getElementById("category_area");
+  const categoryWrapper = document.createElement("div");
+  categoryWrapper.id = "category_wrapper";
+
+  categoryArea.innerHTML = "";
+
+  categoryList.forEach((category) => {
+    const categoryBox = document.createElement("div");
+    categoryBox.id = "category_box";
+
+    const categoryTitle = document.createElement("h3");
+    categoryTitle.innerText = categoryNames[category];
+
+    const deleteIconWrapper = document.createElement("div");
+    deleteIconWrapper.id = "delete_icon_wrapper";
+
+    deleteIconWrapper.addEventListener("click", () => {
+      deleteSelectedCategory(category);
+    });
+
+    const deleteIcon = document.createElement("i");
+    deleteIcon.className = "fa-solid fa-circle-xmark";
+
+    deleteIconWrapper.appendChild(deleteIcon);
+    categoryBox.appendChild(categoryTitle);
+    categoryBox.appendChild(deleteIconWrapper);
+
+    categoryWrapper.appendChild(categoryBox);
+
+    categoryArea.appendChild(categoryWrapper);
+  });
+
+  categoryButton.forEach((button) => {
+    const category = button.value;
+    if (categoryList.includes(category)) {
+      button.checked = true;
+    } else {
+      button.checked = false;
+    }
+  });
```

Comment:
```suggestion
  categoryButton.forEach((button) => {
    button.checked = categoryList.includes(button.value);
  });
```

개인차이지만 요렇게 쓰면 좀 더 간결하게 나타낼 수 있겠당!! ㅎㅎ

Reply:
오 이렇게 써볼 생각은 전혀 못했다..! 고마워👍🏼